src/geo.c

```diff
@@ -699,10 +700,20 @@ void georadiusGeneric(client *c, int srcKeyIndex, int flags) {
     long option_length = 0;
 
     /* Process [optional] requested sorting */
-    if (sort == SORT_ASC) {
-        qsort(ga->array, result_length, sizeof(geoPoint), sort_gp_asc);
-    } else if (sort == SORT_DESC) {
-        qsort(ga->array, result_length, sizeof(geoPoint), sort_gp_desc);
+    if (sort != SORT_NONE) {
+        int (*sort_gp_callback)(const void *a, const void *b) = NULL;
+        if (sort == SORT_ASC) {
+            sort_gp_callback = sort_gp_asc;
+        } else if (sort == SORT_DESC) {
```

Comment:
Technically, this could be else rather than else if, which would prevent sort_gp_callback from ever being NULL should any future enumeration for sort be introduced and the person not catch it in this block of code - though, that crash would obviously be caught by the tests. Not sure from a coding style how you'd like to handle it, so I kept it specific.